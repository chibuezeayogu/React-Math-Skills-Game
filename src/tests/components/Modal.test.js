/* 
global jest 
expect 
*/
import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../../components/Modal';

describe('Modal Component', () => {
  const wrapper = shallow(<Modal />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
