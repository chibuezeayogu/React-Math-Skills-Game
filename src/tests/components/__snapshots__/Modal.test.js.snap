// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render correctly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  id="modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header text-center"
      >
        <h3
          className="modal-title"
          id="exampleModalLabel"
        >
          How To Play
        </h3>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="row text-center"
        >
          The aim of the game is to select all the numbers displayed on the card to match the number of randomly generated stars on the left.
        </div>
        <hr />
        <div
          className="row text-center"
        >
          You can select the exact or any possible combinations of number 
             that sums to the number of stars displayed. Check if your selection 
             is correct by clicking button with equal {=} sign.
        </div>
        <hr />
        <div
          className="row text-center"
        >
          The button is marked green for correct answers and red for wrong 
            answers. You can accept a correct answer by clicking the button 
            again to start another round. Used numbers will no longer 
            be available for selection once answer is accepted
        </div>
        <hr />
        <div
          className="row text-center"
        >
          You are provided with 5 redraws shown on the yellow button. 
            You may use this to generate another set of random stars when the 
            numbers available for selection cannot combine to satisfy 
            the number of stars displayed. 
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
